secure_message/resources/threads.py

```diff
@@ -20,14 +22,73 @@ def get(thread_id):
         logger.info("Getting messages from thread", thread_id=thread_id, user_uuid=g.user.user_uuid)
 
         conversation = Retriever().retrieve_thread(thread_id, g.user)
+        conversation_metadata = Retriever.retrieve_conversation_metadata(thread_id)
 
         logger.info("Successfully retrieved messages from thread", thread_id=thread_id, user_uuid=g.user.user_uuid)
         messages = []
         for message in conversation.all():
             msg = message.serialize(g.user, body_summary=False)
             messages.append(msg)
+
+        if conversation_metadata.is_closed:
+            return jsonify({"messages": add_users_and_business_details(messages),
+                            "is_closed": conversation_metadata.is_closed,
+                            "closed_by": conversation_metadata.closed_by,
+                            "closed_by_uuid": conversation_metadata.closed_by_uuid,
+                            "closed_at": conversation_metadata.closed_at})
+
         return jsonify({"messages": add_users_and_business_details(messages)})
 
+    @staticmethod
+    def patch(thread_id):
+        """Modify conversation metadata"""
+        bound_logger = logger.bind(thread_id=thread_id, user_uuid=g.user.user_uuid)
+        bound_logger.info("Getting metadata for thread")
+        if request.headers['Content-Type'].lower() != 'application/json':
+            bound_logger.info('Request must set accept content type "application/json" in header.')
+            raise BadRequest(description="'application/json' content type in header missing")
+
+        bound_logger.info("Attempting to modify metadata for thread")
+        request_data = request.get_json()
+        metadata = Retriever.retrieve_conversation_metadata(thread_id)
+        ThreadById._validate_request(request_data, metadata)
+
+        if 'is_closed' in request_data:
+            if request_data.get('is_closed'):
+                bound_logger.info("About to close conversation")
+                Modifier.close_conversation(metadata, g.user)
+            else:
+                bound_logger.info("About re-open conversation")
+                Modifier.open_conversation(metadata, g.user)
```

Comment:
Same comment for closing, what should happen if this fails?

Reply:
Returns a 500 now